unyt/array.py

```diff
@@ -164,7 +164,21 @@ def _multiply_units(unit1, unit2):
     return ret.as_coeff_unit()
 
 
+TEMPERATURE_WARNING = """
+After release x, kelvin and Rankine will be temperature deltas in operations with
+heterogeneous units. We recommend using delta_degC and delta_degF to clarify intent.
+"""
```

Comment:
I think it would be better to just error out in the future when we make the behavior change instead of silently changing results. I think it would be better if we encourage our users to write non-ambiguous expressions instead of allowing them in a different way when we do make this behavior change.

Also instead of saying that we'll change behavior in a specific future release I think it would be safer to just leave the exact version when the behavior change will happen ambiguous (e.g. instead of `After release x` I'd say `In the future`). That way we don't confuse anyone unnecessarily if we forget to actually do the behavior change.

Reply:
Addressed both concerns and the behavior will appear as:

```
>>> from unyt import K, degC, delta_degC
>>> 10*K + 23*degC
.../array.py:1766: FutureWarning: Ambiguous operation with heterogeneous temperature units.
In the future, such operations will generate UnitOperationError.
Use delta_degC or delta_degF to avoid the ambiguity.
  mul, unit = unit_operator(u0, u1)
unyt_quantity(33, 'K')
>>> 10*delta_degC + 23*degC
unyt_quantity(33, 'degC')
```